src/mono/wasm/build/WasmApp.targets

```diff
@@ -308,12 +310,12 @@
 
   <Target Name="_GetWasmGenerateAppBundleDependencies">
     <PropertyGroup>
-      <WasmIcuDataFileName Condition="'$(InvariantGlobalization)' != 'true'">icudt.dat</WasmIcuDataFileName>
-
       <_HasDotnetWasm Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.wasm'">true</_HasDotnetWasm>
       <_HasDotnetJsWorker Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.worker.js'">true</_HasDotnetJsWorker>
       <_HasDotnetJsSymbols Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.js.symbols'">true</_HasDotnetJsSymbols>
       <_HasDotnetJs Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.js'">true</_HasDotnetJs>
+      <_WasmIcuFileName Condition="'$(WasmIcuFileName)' != '' and Exists('$(WasmIcuFileName)')">$(WasmIcuFileName)</_WasmIcuFileName>
+      <_WasmIcuFileName Condition="'$(WasmIcuFileName)' != '' and !Exists('$(WasmIcuFileName)')">$(MicrosoftNetCoreAppRuntimePackRidNativeDir)$(WasmIcuFileName)</_WasmIcuFileName>
```

Comment:
If the file is not found in the runtime pack, then we should raise an error saying that we couldn't find the file by the property value itself, and couldn't find it in the runtime pack either. And list the available files from the runtime pack.